File: js/therapy-v3/src/types/appointment.ts
export type AppointmentStatus =
  | "pending"
  | "confirmed"
  | "cancelled"
  | "completed"
  | "no_show";

export interface Appointment {
  id: number;
  patient_id: number;
  specialty_id: number;
  start_at: string;
  end_at: string;
  status: AppointmentStatus;
  notes: string | null;
  admin_notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface AppointmentCreateInput {
  patient_full_name: string;
  patient_phone: string;
  patient_email?: string | null;
  specialty_id: number;
  start_at: string;
}

export interface AppointmentUpdateInput {
  status?: AppointmentStatus;
  notes?: string | null;
  admin_notes?: string | null;
  confirmed_by?: number | null;
}

File: python/therapy-api/src/therapy/appointment/domain/model/appointment.py
from dataclasses import dataclass
from datetime import datetime
from uuid import UUID

from therapy.appointment.domain.model.appointment_status import AppointmentStatus


@dataclass(frozen=True)
class Appointment:
    patient_id: int
    specialty_id: int
    start_at: datetime
    end_at: datetime
    id: int = 0
    status: AppointmentStatus = AppointmentStatus.PENDING
    confirmation_token: UUID | None = None
    cancel_token: UUID | None = None
    notes: str | None = None
    admin_notes: str | None = None
    confirmed_by: int | None = None
    created_at: datetime | None = None
    updated_at: datetime | None = None
